### PR TITLE
feat: don't show confirmation dialog

### PR DESCRIPTION
**Pull request by dariakoko**
The type of this PR is: **Fix**

This PR solves [CX-2870]

### Description

Don't show confirmation dialog when the user has not made any changes | edit/create artwork flow

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CX-2870]: https://artsyproduct.atlassian.net/browse/CX-2870?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx [hunk 1]
```diff
@@ -253,7 +256,16 @@ export const MyCollectionArtworkForm: React.FC<MyCollectionArtworkFormProps> = (
...
                                 as={RouterLink}
-                                onClick={() => setShouldShowBackModal(true)}
+                                onClick={() => {
+                                  const formIsDirty = !isEqual(
+                                    initialValues,
+                                    values
+                                  )
+
+                                  return formIsDirty
+                                    ? setShouldShowBackModal(true)
```
Comment (MounirDhahri) [inline src/Apps/MyCollection/Routes/EditArtwork/MyCollectionArtworkForm.tsx:266]: **suggestion:** I know this was done in a different PR, but I feel like we improve the naming here. Maybe something like `showLeaveWithoutSavingModal` or `showQuitWithoutSavingModal`

File: src/Apps/MyCollection/Routes/EditArtwork/__tests__/MyCollectionArtworkForm.jest.tsx [hunk 2]
```diff
@@ -421,6 +450,19 @@ describe("Create artwork", () => {
         pathname: "/settings/my-collection",
       })
     })
+
+    it("navigates to the previous screen when the form has not been changed", async () => {
```
Comment (MounirDhahri) [inline src/Apps/MyCollection/Routes/EditArtwork/__tests__/MyCollectionArtworkForm.jest.tsx:454]: **praise:** 👏👏👏

File: src/Apps/MyCollection/Routes/EditArtwork/__tests__/MyCollectionArtworkForm.jest.tsx [hunk 3]
```diff
@@ -408,9 +430,16 @@ describe("Create artwork", () => {
   })
 
   describe("Back Link behavior", () => {
-    it("opens modal on press", async () => {
+    it("opens modal on press when the form has been changed", async () => {
```
Comment (MounirDhahri) [inline src/Apps/MyCollection/Routes/EditArtwork/__tests__/MyCollectionArtworkForm.jest.tsx:433]: **praise:** 👏

**Review by MounirDhahri — DISMISSED** [review]
LGTM, I left a small non-blocking comment though that would be great if we address it

**Review by olerichter00 — COMMENTED** [review]
Have you tried using Formik's render prop [dirty](https://formik.org/docs/api/formik#dirty-boolean) instead? If we could use this prop, we wouldn't have to add additional logic with an equality check.

```
dirty: boolean

Returns true if values are not deeply equal from initial values, false otherwise. dirty is a readonly computed property and should not be mutated directly.
```

**Review by MrSltun — DISMISSED** [review]
Great work Daria! 🌟🌟🌟🌟🌟

**Review by olerichter00 — APPROVED** [review]
